plan/eliminate_projection.go

```diff
@@ -84,16 +84,14 @@ func doPhysicalProjectionElimination(p PhysicalPlan) PhysicalPlan {
 // eliminatePhysicalProjection should be called after physical optimization to eliminate the redundant projection
 // left after logical projection elimination.
 func eliminatePhysicalProjection(p PhysicalPlan) PhysicalPlan {
-	oldRoot := p
+	oldSchema := p.Schema()
 	newRoot := doPhysicalProjectionElimination(p)
-	if oldRoot.ID() != newRoot.ID() {
```

Comment:
Why remove it?

Reply:
because `oldRoot.Schema()` may be wrong here. We need cache the schema before project elimination

Reply:
Is there any test to cover it?

Reply:
tidb-test needs right column names, which can cover this case,